src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl

```diff
@@ -75,30 +77,34 @@ References
 
 Shchepetkin, A. F., & McWilliams, J. C. (2005). The regional oceanic modeling system (ROMS): a split-explicit, free-surface, topography-following-coordinate oceanic model. Ocean Modelling, 9(4), 347-404.
 """
-SplitExplicitFreeSurface(FT::DataType = Float64; gravitational_acceleration = g_Earth, kwargs...) = 
-    SplitExplicitFreeSurface(nothing, nothing, nothing, convert(FT, gravitational_acceleration),
-                             SplitExplicitSettings(FT; gravitational_acceleration, kwargs...))
-                             
+function SplitExplicitFreeSurface(grid;
```

Comment:
so now it is not possible anymore to call
`SpliExplicitFreeSurface()`?

Reply:
You need to provide a grid. Is this a problem?

Reply:
I guess it's ok. For the other free surfaces we do not provide a grid but "regularize" with a grid in the model constructor. I see two options: (1) all free surfaces are regularized in the model (2) all free surfaces accept grid as an argument. I am ok with both.

Reply:
Can you point to an example of such regularization? Is it this, for example?

https://github.com/CliMA/Oceananigans.jl/blob/adc63328085687a1908f1c96850aa106311c0d0e/src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl#L172

Reply:
The thing is that SplitExplicit requires grid (in some case, e.g., when cfl is provided); other free surfaces don't require it ever, right?

Reply:
yep. For example, the `ImplicitFreeSurface` is regularized with the method and the grid. So that the `ImplicitFreeSurface` is empty at construction and gets filled in with the correct eta, implicit solver etc at construction. The same for `ExplicitFreeSurface`.

Reply:
Done. I did (1) from what you suggested @simone-silvestri